src/components/range-slider/range-slider.tsx

```diff
@@ -0,0 +1,130 @@
+import classNames from 'classnames';
+import styles from './range-slider.module.scss';
+
+interface RangeSlider {
+    startValue: number;
+    endValue: number;
+    onStartValueChange: (value: number) => void;
+    onEndValueChange: (value: number) => void;
+    /**
+     * The lowest permitted value.
+     */
+    minValue: number;
+    /**
+     * The highest permitted value.
+     */
+    maxValue: number;
+    /**
+     * The granularity that the values must adhere to. @default 1
+     */
+    step?: number;
+    /**
+     * Allows to format the displayed start and end values. For example, add a currency symbol,
+     * format with the specified number of decimal places, etc.
+     */
+    formatValue?: (value: number) => string;
+    startInputName?: string;
+    endInputName?: string;
+}
+
+/**
+ * A slider component for selecting a numeric range.
+ */
+export const RangeSlider = ({
+    startValue,
+    endValue,
+    onStartValueChange,
+    onEndValueChange,
+    minValue,
+    maxValue,
+    step,
+    formatValue = (value) => value.toString(),
+    startInputName,
+    endInputName,
+}: RangeSlider) => {
+    const handleStartValueChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+        const newStartValue = Number(event.target.value);
+        if (newStartValue <= endValue) {
+            onStartValueChange(newStartValue);
+        }
+    };
+
+    const handleEndValueChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+        const newEndValue = Number(event.target.value);
+        if (newEndValue >= startValue) {
+            onEndValueChange(newEndValue);
+        }
+    };
+
+    const handleChangeByClickingOnTrack = (event: React.ChangeEvent<HTMLInputElement>) => {
+        const value = Number(event.target.value);
+        // Change the start or end value
+        // depending on which one is closer to the clicked value on the track.
+        if (Math.abs(value - startValue) < Math.abs(value - endValue)) {
+            onStartValueChange(value);
+        } else {
```

Comment:
If the min and max are equal, we should choose the one that can be adjusted.

Ideally, we should also track which one was selected and avoid switching to the other during a single drag.

Reply:
> Ideally, we should also track which one was selected and avoid switching to the other during a single drag.

I don't know how to reproduce switching to the other during a single drag.

Reply:
> I don't know how to reproduce switching to the other during a single drag.

Here I modified the style of one of the thumbs to show it more clearly. It's not worth fixing, since it requires introduction of a state variable and care should be taken to reset it correctly.

https://github.com/user-attachments/assets/d270a0d9-da0d-4bb9-bba5-7c400e6a42a0